keps/sig-node/4381-dra-structured-parameters/README.md

```diff
@@ -1462,225 +1278,309 @@ prevented. Deleting the entire cluster also leaves resources allocated outside
+    //
+    // There can be at most 32 such reservations. This may get increased in
+    // the future, but not reduced.
+    //
+    // +listType=map
+    // +listMapKey=uid
+    // +patchStrategy=merge
+    // +patchMergeKey=uid
+    // +optional
+    ReservedFor []ResourceClaimConsumerReference
 }
 ```
 
-NamedResourcesFilter and NamedResourcesRequest currently have the same
-content. Despite that, they are defined as separate structs because that might
-change in the future.
-
-
 ##### Allocation result
 
 ```go
 // AllocationResult contains attributes of an allocated resource.
 type AllocationResult struct {
-    // ResourceHandles contain the state associated with an allocation that
-    // should be maintained throughout the lifetime of a claim. Each
-    // ResourceHandle contains data that should be passed to a specific kubelet
-    // plugin once it lands on a node.
+    // ControllerName is the name of the DRA driver which handled the
+    // allocation. That driver is also responsible for deallocating the
+    // claim. It is empty when the claim can be deallocated without
+    // involving a driver.
     //
-    // Setting this field is optional. It has a maximum size of 32 entries.
-    // If null (or empty), it is assumed this allocation will be processed by a
-    // single kubelet plugin with no ResourceHandle data attached. The name of
-    // the kubelet plugin invoked will match the DriverName set in the
-    // ResourceClaimStatus this AllocationResult is embedded in.
+    // A driver may allocate devices provided by other drivers, so this
+    // driver name here can be different from the driver names listed in
+    // the results.
+    //
+    // This is an alpha field and requires enabling the DRAControlPlaneController
+    // feature gate.
+    //
+    // +optional
+    ControllerName string
+
+    // This field holds configuration for all drivers which
+    // satisfied requests in this claim. The configuration applies to
+    // the entire claim.
+    //
+    // +optional
+    // +listType=atomic
+    Config []ConfigurationParameters
+
+    // Results lists all allocated devices.
     //
     // +listType=atomic
-    ResourceHandles []ResourceHandle
+    Results []RequestAllocationResult
 
-    // This field will get set by the resource driver after it has allocated
-    // the resource to inform the scheduler where it can schedule Pods using
-    // the ResourceClaim.
+    // Setting this field is optional. If unset, the allocated devices are available everywhere.
     //
-    // Setting this field is optional. If null, the resource is available
-    // everywhere.
     // +optional
-    AvailableOnNodes *core.NodeSelector
+    AvailableOnNodes *v1.NodeSelector
 
-    // Shareable determines whether the resource supports more
-    // than one consumer at a time.
+    // Shareable determines whether the claim supports more than one consumer at a time.
+    //
     // +optional
     Shareable bool
```

Comment:
Would we drop this as well in light of:
https://github.com/kubernetes/enhancements/pull/4709/files#r1632269363

Reply:
Done.